source/client-side-encryption/tests/README.rst

```diff
@@ -0,0 +1,208 @@
+============================
+Client Side Encryption Tests
+============================
+
+.. contents::
+
+----
+
+Introduction
+============
+
+This document describes the format of the driver spec tests included in the JSON
+and YAML files included in this directory.
+
+Additional prose tests, that are not represented in the spec tests, are described
+and MUST be implemented by all drivers.
+
+TODO
+----
+Once limitations of query parsing in mongocryptd are known, add tests to exercise those limitations.
+
+Spec Test Format
+================
+
+The spec tests are closely based on the transactions spec tests and CRUD spec tests, with the following notable differences:
+
+- An ``encryption_setup`` section, describing client side encryption setup.
+
+- The addition of ``clientSideEncryptionOptions`` on clients.
+
+- The addition of the per-test ``kmsFailpoint`` and ``mongocryptdFailpoint`` to control the mock KMS and mongocryptd servers.
+
+- Operations may be specified per collection through an optional "object" field. I.e. operations use the default collection specified at the root of the test:
+
+  .. code:: json
+
+     {
+         "name": "insertOne",
+         "documents": [ { "_id": 1 } ]
+     }
+
+  inserts on the default collection, but
+
+  .. code:: json
+
+      {
+          "name": "insertOne",
+          "object": { "database": "db", "collection": "other" }
```

Comment:
I think this section is still outdated given the recent changes to `object`.

Reply:
Fixed.